core/src/main/java/org/apache/iceberg/LocationProviders.java

```diff
@@ -167,5 +169,17 @@ private static String pathContext(String tableLocation) {
 
       return resolvedContext;
     }
+
+    private static String computeHash(String fileName) {
+      Preconditions.checkState(fileName != null, "fileName cannot be null");
+      byte[] hashBuffer = new byte[6];
```

Comment:
Is this buffer necessary? Seems better to avoid unnecessary allocations.

Reply:
As per my understanding, this is required, since we generate 4 bytes hash, and to pass it to base64 encode we need size to be a multiple of 3 .. hence we create 6 size, byte array (as you suggested above), one possible option is to extract only 3 bytes from hash and do it without using extra memory (still thinking through). please let me take a deeper look, if it is suff and can be done without taking any extra memory !

---

One observation, a long variable would also have taken 8 bytes and since this is also a local variable, thinking if it will add some memory pressure in some scenario.